packages/flutter_tools/lib/src/commands/migrate_apply.dart

```diff
@@ -0,0 +1,205 @@
+            '`$kDefaultMigrateWorkingDirectoryName`',
+      valueHelp: 'path',
+    );
+    argParser.addOption(
+      'project-directory',
+      help: 'The root directory of the flutter project. This defaults to the '
+            'current working directory if omitted.',
+      valueHelp: 'path',
+    );
+    argParser.addFlag(
+      'force',
+      abbr: 'f',
+      help: 'Ignore unresolved merge conflicts and apply staged changes by force.',
+    );
+    argParser.addFlag(
+      'keep-working-directory',
+      help: 'Do not delete the working directory.',
+    );
+  }
+
+  final bool _verbose;
+
+  final Logger logger;
+
+  final FileSystem fileSystem;
+
+  final Terminal terminal;
+
+  final MigrateUtils migrateUtils;
+
+  @override
+  final String name = 'apply';
+
+  @override
+  final String description = r'Accepts the changes produced by `$ flutter '
+                              'migrate start` and copies the changed files into '
+                              'your project files. All merge conflicts should '
+                              'be resolved before apply will complete '
+                              'successfully. If conflicts still exist, this '
+                              'command will print the remaining conflicted files.';
+
+  @override
+  String get category => FlutterCommandCategory.project;
+
+  @override
+  Future<Set<DevelopmentArtifact>> get requiredArtifacts async => const <DevelopmentArtifact>{};
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
+    final String? projectDirectory = stringArg('project-directory');
+    final FlutterProjectFactory flutterProjectFactory = FlutterProjectFactory(logger: logger, fileSystem: fileSystem);
+    final FlutterProject project = projectDirectory == null
+      ? FlutterProject.current()
+      : flutterProjectFactory.fromDirectory(fileSystem.directory(projectDirectory));
+
+    if (!await gitRepoExists(project.directory.path, logger, migrateUtils)) {
+      logger.printStatus('No git repo found. Please run in a project with an '
+                         'initialized git repo or initialize one with:');
+      printCommandText('git init', logger);
+      return const FlutterCommandResult(ExitStatus.fail);
+    }
+
+    final bool force = boolArg('force') ?? false;
+
+    terminal.usesTerminalUi = true;
+
+    Directory workingDirectory = project.directory.childDirectory(kDefaultMigrateWorkingDirectoryName);
+    final String? customWorkingDirectoryPath = stringArg('working-directory');
+    if (customWorkingDirectoryPath != null) {
+      if (fileSystem.path.isAbsolute(customWorkingDirectoryPath)) {
+        workingDirectory = fileSystem.directory(customWorkingDirectoryPath);
+      } else {
+        workingDirectory = project.directory.childDirectory(customWorkingDirectoryPath);
+      }
+    }
+    if (!workingDirectory.existsSync()) {
+      logger.printStatus('No migration in progress. Please run:');
```

Comment:
can you interpolate `workingDirectory` in case the problem is that the tool is looking in the wrong place?